packages/models-library/src/models_library/services_resources.py

```diff
@@ -43,11 +43,16 @@ def ensure_limits_are_equal_or_above_reservations(cls, values):
         if isinstance(values["reservation"], str):
             # in case of string, the limit is the same as the reservation
             values["limit"] = values["reservation"]
-        else:
+        elif values["limit"] <= 0:
             values["limit"] = max(values["limit"], values["reservation"])
+        elif values["limit"] < values["reservation"]:
```

Comment:
ok, so docker handles both types, either int or strings (like version="2.3.3" for example or Memory=2048)
the float I added because instead of NanoCPUs we use CPUs.